integration_test.go

```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
 	// TODO(oskar) - gracefully stop proxy by the end of the test
 	es.Stop(ctx)
 }
+
+func TestBasicAuthRoundtrip(t *testing.T) {
+	tc := newTestClient()
+
+	echoServerAddr := "localhost:3010"
+	es := newEchoServer(echoServerAddr)
+
+	ctx := context.Background()
+
+	payload := fmt.Sprintf("{%q:%q}", "hello", "world")
+	payloadBytes := []byte(payload)
+
+	go es.Start()
+
+	// Authentication Proxy setup
+	username := "username"
+	password := "password"
+
+	authProxyConfig := ProxyConfig{
+		RemoteAddress:              "http://" + echoServerAddr,
+		ShouldValidateRequests:     true,
+		AuthenticationScheme:       BasicAuthScheme,
+		AllowedBasicAuthUserString: fmt.Sprintf("%s:%s", username, password),
+	}
+	authProxyAddr := "localhost:3012"
+
+	proxy, err := NewProxy(&authProxyConfig)
+
+	require.NoError(t, err)
+
+	go func() {
+		t.Log(http.ListenAndServe(authProxyAddr, proxy))
+	}()
+
+	// Signign Proxy Setup
+	signingProxyConfig := ProxyConfig{
+		RemoteAddress:      "http://" + authProxyAddr,
+		ShouldSignOutgoing: true,
+		SigningScheme:      BasicAuthScheme,
+		BasicAuthUser:      username,
+		BasicAuthPassword:  password,
+	}
+	signingProxyAddr := "localhost:3013"
+
+	signingProxy, err := NewProxy(&signingProxyConfig)
+
+	require.NoError(t, err)
```

Comment:
Ditto above, no need for newline between call and error check.

Reply:
Please fix everywhere.

Reply:
Done for all.